PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/MavenUtils.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.util;
+
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.common.exceptions.AzureExecutionException;
+import com.microsoft.intellij.maven.SpringCloudDependencyManager;
+import org.apache.commons.lang3.StringUtils;
+import org.dom4j.DocumentException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.idea.maven.model.MavenExplicitProfiles;
+import org.jetbrains.idea.maven.project.MavenEmbeddersManager;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+import org.jetbrains.idea.maven.project.MavenProjectsProcessor;
+import org.jetbrains.idea.maven.server.MavenEmbedderWrapper;
+
+import java.nio.file.Paths;
+import java.util.concurrent.*;
+
+public class MavenUtils {
+    private static final ExecutorService THREAD_POOL = Executors.newCachedThreadPool();
+
+    public static String getSpringBootFinalJarPath(@NotNull Project ideaProject, @NotNull MavenProject mavenProject,
+                                                   int timeoutSeconds)
+            throws AzureExecutionException, DocumentException {
+        String xml = evaluateEffectivePom(ideaProject, mavenProject, timeoutSeconds);
+        if (StringUtils.isEmpty(xml)) {
+            throw new AzureExecutionException("Failed to evaluate effective pom for project: " + ideaProject.getName());
+        }
+        SpringCloudDependencyManager manager = new SpringCloudDependencyManager(xml);
+        String finalName = manager.getPluginConfiguration("org.springframework.boot", "spring-boot-maven"
```

Comment:
Shall we check whether `spring-boot-maven-plugin` exists here? Or it will be done in configuration validation part?

Reply:
No need by now, if it is here, then we will use the final name in spring-boot-maven-plugin, if not, we will use final name